src/user_interface/menu.rs

```diff
@@ -72,96 +116,111 @@ impl Menu {
                     _ => {}
                 }
             }
-            unique
-                .iter()
-                .map(|first| {
-                    let mut new_below = below.clone();
-                    new_below.push(first.clone());
-                    Menu::submenu(first.as_str(), Menu::fabric_menu(fabrics, new_below))
-                })
-                .collect()
+            let mut menu = menu;
+            for first in unique {
+                let mut new_below = below.clone();
+                new_below.push(first.clone());
+                menu = menu.submenu(ALWAYS, Menu::fabric_menu_recurse(Menu::new(first.as_str()), fabrics, new_below));
+            }
+            menu
         } else {
-            sub_fabrics
-                .into_iter()
-                .map(|fabric_plan| {
-                    let label = fabric_plan.name.last().unwrap();
-                    Menu::action(label.as_str(), Action::Crucible(CrucibleAction::BuildFabric(fabric_plan.clone())))
-                })
-                .collect()
+            let mut menu = Menu::new(below.last().unwrap());
+            for fabric_plan in sub_fabrics {
+                let label = fabric_plan.name.last().unwrap();
+                menu = menu.action(
+                    label.as_str(), false, ALWAYS,
+                    Action::Crucible(CrucibleAction::BuildFabric(fabric_plan.clone())),
+                );
+            }
+            menu
         }
     }
 
-    fn speed_menu() -> Vec<Menu> {
-        [(0usize, "Paused"), (5, "Glacial"), (25, "Slow"), (125, "Normal"), (625, "Fast")]
-            .into_iter()
-            .map(|(speed, label)|
-                Menu::action(label, Action::Crucible(CrucibleAction::SetSpeed(speed))))
-            .collect()
+    fn fabric_menu(fabrics: &[FabricPlan]) -> Menu {
+        Self::fabric_menu_recurse(Menu::new("Fabrics"), fabrics, Vec::new())
+    }
+
+    fn speed_menu() -> Menu {
+        let mut menu = Menu::new("Speed");
+        for (speed, label) in [(0usize, "Paused"), (5, "Glacial"), (25, "Slow"), (125, "Normal"), (625, "Fast")] {
+            menu = menu.action(label, true, |_| true, Action::Crucible(CrucibleAction::SetSpeed(speed)));
```

Comment:
```suggestion
            menu = menu.action(label, true, ALWAYS, Action::Crucible(CrucibleAction::SetSpeed(speed)));
```
Unless I misunderstand